lib/QubitInformationObjectXmlCache.class.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Export information object(s) as EAD and/or DC XML.
+ *
+ * @package    AccesstoMemory
+ * @subpackage library
+ * @author     Mike Cantelon <mike@artefactual.com>
+ */
+class QubitInformationObjectXmlCache
+{
+  public function __construct()
+  {
+    $this->createExportDestinationDirs();
+  }
+
+  /**
+   * Check if downloads/exports, and format-specific subdirectories, have been created and, if not,
+   * create them.
+   *
+   * @return null
+   */
+  protected function createExportDestinationDirs()
+  {
+    $exportsPath = sfConfig::get('sf_web_dir') . DIRECTORY_SEPARATOR . 'downloads' . DIRECTORY_SEPARATOR .'exports';
+
+    $eadExportPath = $exportsPath . DIRECTORY_SEPARATOR .'ead';
+    if (!is_dir($eadExportPath))
+    {
+      mkdir($eadExportPath, 0755, true);
+    }
+
+    $dcExportPath = $exportsPath . DIRECTORY_SEPARATOR .'dc';
+    if (!is_dir($dcExportPath))
+    {
+      mkdir($dcExportPath, 0755, true);
+    }
+  }
+
+  /**
+   * Export information object to EAD (if top-level) and DC.
+   *
+   * @param object  information object to be export
+   *
+   * @return null
+   */
+  public function export($resource)
+  {
+    // Only cache top-level information object's EAD XML
+    if ($resource->parentId == QubitInformationObject::ROOT_ID)
+    {
+      $this->cacheXmlRepresentation($resource, 'ead');
+    }
+
+    $this->cacheXmlRepresentation($resource, 'dc');
```

Comment:
this would seem to get called even if we're dealing with EAD and the if evaluates true above here, right? is that desirable?